### PR TITLE
Soko2 fixed suggestion logic, and added .catch() in posts.js

### PR DESCRIPTION
**Pull request by solongoch**

### CODE REVIEW

File: routes/api/posts.js [hunk 1]
```diff
@@ -147,7 +148,7 @@ router.delete(
           // delete ':id' post
           post.remove().then(() => res.json({ success: true , message: "Post deleted" }));
         })
-        .catch(err => res.status(404).json({ postnotfound: "No post found" }));
+        .catch(err => res.status(404).json({success: false, error: err.message}));
```
Comment (slpriya) [inline routes/api/posts.js]: status code 500 and else for post not found

**Review by Emily484 — APPROVED** [review]
looks good

File: routes/api/posts.js [hunk 2]
```diff
@@ -124,7 +124,8 @@ router.get("/:id", accessRouteWithOrWithoutToken, (req, res) => {
       } else { // accessing post from private account and user not logged in
         res.send("This is a private account! Please log in.");
       }
-    });
+    })
```
Comment (slpriya) [inline routes/api/posts.js:138]: Status Code 500 and else for post not found.

**Review by slpriya — CHANGES_REQUESTED** [review]

**Review by slpriya — APPROVED** [review]

**Review by Emily484 — APPROVED** [review]
looks good

**Pull request merged**